crates/turborepo-lib/src/engine/builder.rs

```diff
@@ -153,6 +153,28 @@ impl<'a> EngineBuilder<'a> {
         self
     }
 
+    // Returns the set of allowed tasks that can be run if --only is used
+    // The set is exactly the product of the packages in filter and tasks specified
+    // by CLI
+    fn allowed_tasks(&self) -> Option<HashSet<TaskId<'static>>> {
+        if self.tasks_only {
+            Some(
+                self.workspaces
+                    .iter()
+                    .cartesian_product(self.tasks.iter())
+                    .map(|(package, task_name)| {
+                        task_name
+                            .task_id()
+                            .unwrap_or(TaskId::new(package.as_ref(), task_name.task()))
+                            .into_owned()
+                    })
+                    .collect(),
+            )
+        } else {
+            None
```

Comment:
Why is this none? shouldn't it be all?

Reply:
never mind, I saw the rest of the code and saw how allowed_tasks is used. the name of the method just threw me off